sdk/core/Azure.Core.Amqp/src/AmqpMessageProperties.cs

```diff
@@ -14,43 +14,22 @@ public class AmqpMessageProperties
         /// <summary>
         /// Initializes a new <see cref="AmqpMessageProperties"/> instance.
         /// </summary>
-        public AmqpMessageProperties() { }
-
-        /// <summary>
-        /// Initializes a new <see cref="AmqpMessageProperties"/> instance by copying the passed in properties.
-        /// </summary>
-        /// <param name="properties">The properties to copy.</param>
-        public AmqpMessageProperties(AmqpMessageProperties properties)
-        {
-            MessageId = properties.MessageId;
-            UserId = properties.UserId;
-            To = properties.To;
-            Subject = properties.Subject;
-            ReplyTo = properties.ReplyTo;
-            CorrelationId = properties.CorrelationId;
-            ContentType = properties.ContentType;
-            ContentEncoding = properties.ContentEncoding;
-            AbsoluteExpiryTime = properties.AbsoluteExpiryTime;
-            CreationTime = properties.CreationTime;
-            GroupId = properties.GroupId;
-            GroupSequence = properties.GroupSequence;
-            ReplyToGroupId = properties.ReplyToGroupId;
-        }
+        internal AmqpMessageProperties() { }
 
         /// <summary>
         /// The message-id value from the AMQP properties.
         /// </summary>
-        public string? MessageId { get; set; }
+        public AmqpMessageId? MessageId { get; set; }
 
         /// <summary>
         /// The user-id value from the AMQP properties.
         /// </summary>
-        public BinaryData? UserId { get; set; }
+        public ReadOnlyMemory<byte>? UserId { get; set; }
```

Comment:
Did we stop using BinaryData here for some other reason other than the type being a class?

Reply:
There was a comment in the API review that it isn't needed here as this is a low level API and we should focus more on perf than on usability.